layouts/shortcodes/dbm-mysql-agent-config-examples.md

```diff
@@ -0,0 +1,95 @@
+In these cases, it is recommended to limit to the number of instances per Agent to a maximum of 10 database instances to guarantee reliable performance.
+
+```yaml
+init_config:
+instances:
+  - dbm: true
+    host: products-primary.123456789012.us-east-1.rds.amazonaws.com
+    port: 3306
+    username: datadog
+    password: '<PASSWORD>'
+    tags:
+      - 'env:prod'
+      - 'team:team-discovery'
+      - 'service:product-recommendation'
+  - dbm: true
+    host: products–replica-1.us-east-1.rds.amazonaws.com
+    port: 3306
+    username: datadog
+    password: '<PASSWORD>'
+    options:
+      replication: true
+    tags:
+      - 'env:prod'
+      - 'team:team-discovery'
+      - 'service:product-recommendation'
+  - dbm: true
+    host: products–replica-2.us-east-1.rds.amazonaws.com
+    port: 3306
+    username: datadog
+    password: '<PASSWORD>'
+    options:
+      replication: true
+    tags:
+      - 'env:prod'
+      - 'team:team-discovery'
+      - 'service:product-recommendation'
+    [...]
+```
+
+### Storing passwords securely
+While it is possible to declare passwords directly in the Agent configuration files, it is a more secure practice to encrypt and store database credentials elsewhere using secret management software such as [Vault](https://www.vaultproject.io/). The Agent is able to read these credentials using the `ENC[]` syntax. Review the [secrets management documentation](/agent/guide/secrets-management/) for the required setup to store these credentials. The following example shows how to declare and use those credentials:
+```yaml
+init_config:
+instances:
+  - dbm: true
+    host: localhost
+    port: 3306
+    username: datadog
+    password: 'ENC[datadog_user_database_password]'
+```
+
+### Running custom queries
+To collect custom metrics, use the `custom_queries` option. See the sample [mysql.d/conf.yaml](https://github.com/DataDog/integrations-core/blob/master/mysql/datadog_checks/mysql/data/conf.yaml.example) for more details.
+```yaml
+init_config:
+instances:
+  - dbm: true
+    host: localhost
+    port: 3306
+    username: datadog
+    password: '<PASSWORD>'
+    custom_queries:
+    - query: SELECT age, salary, hours_worked, name FROM hr.employees;
+      columns:
+        - name: custom.employee_age
+          type: gauge
+        - name: custom.employee_salary
+           type: gauge
+        - name: custom.employee_hours
+           type: count
+        - name: name
+           type: tag
+      tags:
+        - 'table:employees'
+```
+### Working with hosts through a proxy
+If the Agent must connect through a proxy such as the [Cloud SQL Auth proxy](https://cloud.google.com/sql/docs/mysql/connect-admin-proxy), all telemetry is tagged with the hostname of the proxy rather than the database instance. Utilize the `reported_hostname` option to set a custom override of the hostname detected by the Agent.
```

Comment:
```suggestion
If the Agent must connect through a proxy such as the [Cloud SQL Auth proxy](https://cloud.google.com/sql/docs/mysql/connect-admin-proxy), all telemetry is tagged with the hostname of the proxy rather than the database instance. Use the `reported_hostname` option to set a custom override of the hostname detected by the Agent.
```